### PR TITLE
Sed 1789 handling empty package location

### PR DESCRIPTION
**Pull request by jeromebrongniart**

### CODE REVIEW

**Comment by david-stephan** [comment]
@jeromebrongniart small request for change, otherwise it looks OK
@jeromecomte F.y.i

**Comment by jeromebrongniart** [comment]
@david-stephan Any other change than the @JsonIgnore?

**Comment by david-stephan** [comment]
@jeromebrongniart yes, see my comment from Friday, line 213 on FunctionPackageManager

File: step-functions/step-functions-package/src/main/java/step/functions/packages/FunctionPackageManager.java [hunk 1]
```diff
@@ -210,7 +210,7 @@ private List<Function> deleteFunctions(FunctionPackage previousFunctionPackage)
 	private FunctionPackage addOrUpdateFunctionPackage(FunctionPackage previousFunctionPackage,
 			FunctionPackage newFunctionPackage) throws Exception {
 		String packageLocation = newFunctionPackage.getPackageLocation();
-		if (packageLocation == null || packageLocation.trim().length() == 0) {
```
Comment (jeromebrongniart) [inline step-functions/step-functions-package/src/main/java/step/functions/packages/FunctionPackageManager.java:213]: @david-stephan I cannot see your comment

Comment (david-stephan) [reply step-functions/step-functions-package/src/main/java/step/functions/packages/FunctionPackageManager.java:213]: image

Comment (jeromebrongniart) [reply step-functions/step-functions-package/src/main/java/step/functions/packages/FunctionPackageManager.java:213]: ok, I see the problem and added back a part of the refactoring that was revert by mistake

**Review by david-stephan — APPROVED** [review]

**Comment by jeromebrongniart** [comment]
@david-stephan ready for review:
- I removed the change in the FunctionPackage
- I added the sanity checks at every entry point of the class

**Comment by jeromebrongniart** [comment]
@david-stephan for me it can be merge. Did I miss something?

**Comment by david-stephan** [comment]
> @david-stephan for me it can be merge. Did I miss something?

@jeromebrongniart yes my previous 2 comments :-)  about removing "assert functionPackageId != null;" in FunctionPackageManager line 138

**Comment by jeromebrongniart** [comment]
> > @david-stephan for me it can be merge. Did I miss something?
> 
> @jeromebrongniart yes my previous 2 comments :-) about removing "assert functionPackageId != null;" in FunctionPackageManager line 138

I cannot see your comment :/
- I added back the missing assert
- these asserts are never called. I think they should be changed by proper execption handling. This is the only place we use that with UserAccessorImpl. I open a separete ticket for that

**Review by david-stephan — APPROVED** [review]

**Pull request merged**